Adjust WebGL2RenderingContext::deleteTransformFeedback validation

lexaknyazev opened the pull request:
#### d8119af7ac88f5cd86917520af6ce50343787f3f
<pre>
Adjust WebGL2RenderingContext::deleteTransformFeedback validation
<a href="https://bugs.webkit.org/show_bug.cgi?id=223359">https://bugs.webkit.org/show_bug.cgi?id=223359</a>

Reviewed by Kenneth Russell.

Some checks were bypassed because of
the incorrect context loss condition.

Added an error message when the passed
object belongs to another context.

* LayoutTests/TestExpectations:
* Source/WebCore/html/canvas/WebGL2RenderingContext.cpp:
(WebCore::WebGL2RenderingContext::deleteTransformFeedback):

Canonical link: <a href="https://commits.webkit.org/252372@main">https://commits.webkit.org/252372@main</a>
</pre>

kenrussell reviewed (approved):
Looks good given that this makes the test fully pass.

Should the commit description match the title of the WebKit bug?

Do you plan to revisit other transform feedback-related validation? In particular it looks like useProgram isn't checking to make sure transform feedback isn't active. I didn't look at other entry points.

lexaknyazev committed:
Adjust WebGL2RenderingContext::deleteTransformFeedback validation
https://bugs.webkit.org/show_bug.cgi?id=223359

Reviewed by Kenneth Russell.

Some checks were bypassed because of
the incorrect context loss condition.

Added an error message when the passed
object belongs to another context.

* LayoutTests/TestExpectations:
* Source/WebCore/html/canvas/WebGL2RenderingContext.cpp:
(WebCore::WebGL2RenderingContext::deleteTransformFeedback):

Canonical link: https://commits.webkit.org/252372@main
